command/clouds/aws/ecs.go

```diff
@@ -0,0 +1,170 @@
+	"github.com/innogames/slack-bot/v2/bot/matcher"
+	"github.com/innogames/slack-bot/v2/bot/msg"
+)
+
+// command to trigger/start jenkins jobs
+type ecsCommand struct {
+	awsCommand
+}
+
+var _ecs *ecs.ECS
+
+// NewAwsCommand is a command to interact with aws resources
+func newEcsCommands(base awsCommand) bot.Command {
+	return &ecsCommand{base}
+}
+
+func (c *ecsCommand) GetMatcher() matcher.Matcher {
+	return matcher.NewGroupMatcher(
+		matcher.NewRegexpMatcher(`ecs ls (?P<CLUSTER>.*)`, c.list),
+		matcher.NewRegexpMatcher(`ecs restart (?P<CLUSTER>.*) (?P<SERVICE>.*)`, c.restart),
+	)
+}
+
+// We want to retrieve all services from an ecs cluster and throw them as a slack block
+
+func (c *ecsCommand) list(match matcher.Result, message msg.Message) {
+	// TODO: make it an input
+	cluster := match.GetString("CLUSTER")
+	svc, err := ListServices(cluster)
+	if err != nil {
+		log.Println(err.Error())
+		c.SendEphemeralMessage(message, err.Error())
+		return
+	}
+
+	var text string
+
+	text += "Hello <@" + message.User + ">, current services:. \n"
+
+	for _, v := range svc {
+		text += fmt.Sprintf("• %s", v)
+		text += "\n"
+	}
+
+	// t.SendEphemeralMessage(message, text)
+
+	c.SendEphemeralMessage(message, text)
+}
+
+// restart service
+
+func (c *ecsCommand) restart(match matcher.Result, message msg.Message) {
+	cluster := match.GetString("CLUSTER")
+	svc := match.GetString("SERVICE")
+	log.Println("restart requested: ", svc)
+	err := ForceNewDeployment(cluster, svc)
+
+	if err != nil {
+		log.Println(err.Error())
+		c.SendEphemeralMessage(message, err.Error())
+		return
+	}
+
+	// TODO let user know when restart is completed
+	// message := fmt.Sprintf("restart completed")
+	c.SendMessage(message, svc)
+}
+
+func (c *ecsCommand) GetHelp() []bot.Help {
+	examples := []string{
+		"ecs ls my-cluster-name // to list services in a cluster",
+		"ecs restart my-cluster-name service // to restart a service in a cluster",
+	}
+
+	help := make([]bot.Help, 0)
+	help = append(help, bot.Help{
+		Command:     "ecs <sub command>",
+		Description: "interact with aws ecs resources",
+		Examples:    examples,
+		Category:    category,
+	})
+
+	return help
+}
+
+func ListServices(cluster string) ([]string, error) {
+	svc := assertECS()
+	params := &ecs.ListServicesInput{Cluster: aws.String(cluster)}
+	result := make([]*string, 0)
+	err := svc.ListServicesPages(params, func(services *ecs.ListServicesOutput, lastPage bool) bool {
+		result = append(result, services.ServiceArns...)
+		return !lastPage
+	})
+	if err != nil {
+		return nil, err
+	}
+	out := make([]string, len(result))
+	for i, s := range result {
+		out[i] = path.Base(*s)
+	}
+	log.Println(out)
+	return out, nil
+}
+
+func assertECS() *ecs.ECS {
+	if _ecs == nil {
+		_ecs = ecs.New(session.New(getServiceConfiguration()))
+	}
+	return _ecs
+}
+
+func getServiceConfiguration() *aws.Config {
+	region := os.Getenv("AWS_REGION")
+	if region == "" {
+		region = "us-east-1"
```

Comment:
maybe also add a log entry if no AWS_REGION is set. else I'd expect some debug hell

Reply:
added log.Info couldn't find a way to make it debug